File: dagger2/src/main/java/pgreze/dagger2/ui/player/PlayerFragment.java
package pgreze.dagger2.ui.player;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import javax.inject.Inject;

import pgreze.dagger2.R;
import pgreze.dagger2.di.HasComponent;
import pgreze.dagger2.ui.ActivityComponent;

import static pgreze.dagger2.ui.player.PlayerInteractor.Status;

public class PlayerFragment extends Fragment {

    @Inject PlayerInteractor playerInteractor;
    private Button statusBtn;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ((HasComponent<ActivityComponent>) getActivity()).component().inject(this);
    }

    @Override
    public View onCreateView(LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_player, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        statusBtn = (Button) view.findViewById(R.id.status_btn);
        updateStatusBtn(playerInteractor.getStatus());
        statusBtn.setOnClickListener(v -> {
            Status status = playerInteractor.onIndicatorClick();
            updateStatusBtn(status);
        });
    }

    public void updateStatusBtn(Status status) {
        switch (status) {
            case PLAY:
                statusBtn.setText("Pause");
                break;
            case PAUSE:
            case STOP:
                statusBtn.setText("Play");
                break;
        }
    }
}

File: dagger2/src/main/java/pgreze/dagger2/ui/player/PlayerFragment.kt
package pgreze.dagger2.ui.player

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button

import javax.inject.Inject

import pgreze.dagger2.R
import pgreze.dagger2.di.HasComponent
import pgreze.dagger2.ui.ActivityComponent

import pgreze.dagger2.ui.player.PlayerPresenter.Status

class PlayerFragment : Fragment() {

    @Inject
    internal lateinit var playerPresenter: PlayerPresenter

    private lateinit var statusBtn: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        (activity as HasComponent<ActivityComponent>).component.inject(this)
    }

    override fun onCreateView(
            inflater: LayoutInflater,
            container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View? = inflater.inflate(R.layout.fragment_player, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        statusBtn = view.findViewById<Button>(R.id.status_btn)
        statusBtn.setOnClickListener { v ->
            val status = playerPresenter.onIndicatorClick()
            updateStatusBtn(status)
        }

        updateStatusBtn(playerPresenter.status)
    }

    fun updateStatusBtn(status: Status) {
        when (status) {
            PlayerPresenter.Status.PLAY -> statusBtn.text = "Pause"
            else -> statusBtn.text = "Play"
        }
    }
}
